pkg/controller/registry/reconciler/grpc.go

```diff
@@ -168,6 +169,78 @@ func (c *GrpcRegistryReconciler) ensureService(source grpcCatalogSourceDecorator
 	return err
 }
 
+// updateRegistryPodByDigest is an internal method that verifies the latest catalog source by comparing image digests.
+// If the digests do not match (i.e. there is a new version of the catalog source) then overwrite the catalog source pod.
+// This is done to only have the container runtime (CRI-O) talk to the container registry.
+func (c *GrpcRegistryReconciler) updateRegistryPodByDigest(source grpcCatalogSourceDecorator) bool {
+	podDigestChan := make(chan string, 1)
+	lastCheckedTimestamp := time.Time{}
+	interval := source.Spec.Poll.Interval.Duration
+
+	// check poll value is not zero (default poll value)
+	// if polling interval is zero polling will not be done
+	if interval == time.Duration(0) {
+		return false
+	}
+
+	// check digest every poll interval
+	if time.Now().After(lastCheckedTimestamp.Add(interval)) &&
```

Comment:
If we keep compare the polling time against creation time without anyway to keep track when is the last time we check for the update, then this if statement will return `true` after creation-time + poll-interval. I would add a new field to `CatalogSource` to keep track of `LastUpdateTime` and also a func to determine if the time is up for an update attempt based on Poll. Essentially, that function will trigger this `updateRegistryPodByDigest` as a part of sync `CatalogSource` process that is controlled by catalog operator. See my [PR](https://github.com/Denton24646/operator-lifecycle-manager/pull/1/).